java/vector/src/main/java/org/apache/arrow/vector/stream/MessageSerializer.java

```diff
@@ -120,6 +120,9 @@ public static ArrowBlock serialize(WriteChannel out, ArrowRecordBatch batch)
 
     long start = out.getCurrentPosition();
     int bodyLength = batch.computeBodyLength();
+    if (bodyLength % 8 != 0) {
+      bodyLength += 8 - (bodyLength % 8);
+    }
```

Comment:
I'm not sure if buffers are always a multiple of 8 - I didn't see anything that guaranteed that, but you could be right. I'll add the padding for schema messages.